apps/frontend/core-dapp/src/features/trade/close-trade/CloseTradeStore.ts

```diff
@@ -278,33 +280,69 @@ export class CloseTradeStore {
 
   // fees and rewards
 
-  private get fees(): FeeStore[] | undefined {
+  private get uniswapFeeStore(): FeeStore | undefined {
     const { selectedMarket } = this
     if (selectedMarket === undefined) return undefined
 
-    if (selectedMarket.resolved) {
-      const { redemptionFee, tokenSender } = selectedMarket
-      if (redemptionFee === undefined || tokenSender === undefined) return undefined
-
-      return [
-        new FeeStore({
-          applicableToWstEth: () => this.amountOutInWstEthBN,
-          fee: redemptionFee,
-          tokenSender,
-        }),
-      ]
+    const fee = selectedMarket.resolved
+      ? fraction(BigNumber.from(0), BigNumber.from(1))
+      : TradeStore.UNISWAP_FEE
+
+    const reverseFee = (amountAfterFee: BigNumber | undefined): BigNumber | undefined => {
+      if (amountAfterFee === undefined) return undefined
+
+      return div(
+        amountAfterFee,
+        sub(fraction(BigNumber.from(1), BigNumber.from(1)), TradeStore.UNISWAP_FEE)
+      )
     }
 
+    return new FeeStore({
+      // When quoting, Uniswap gives us the amount after fee. To calculate the
+      // correct fee we have to reverse it and then re-apply it.
+      applicableToWstEth: () => reverseFee(this.amountOutInWstEthBN),
+      fee,
+      tokenSender: undefined,
+    })
+  }
+
+  private get tradeFeeStore(): FeeStore | undefined {
+    const { selectedMarket } = this
+    if (selectedMarket === undefined) return undefined
```

Comment:
It's not obvious why selectedMarket needs to be checked here since it's not used in this function, better to just check for uniswapFeeStore? Also if we're going to check selectedMarket, we dont have to destructure it because the function is not using many things from `this`

```suggestion
    if(this.uniswapFeeStore === undefined) return undefined
```

OR this if we're keeping the check for selectedMarket

```suggestion
    if(this.selectedMarket === undefined) return undefined
```

Reply:
This was just an oversight, there's no need to check for `selectedMarket` (there's also no need to check for `uniswapFeeStore` since we're using optional chaining. Thanks for catching!